astropy/table/operations.py

```diff
@@ -700,6 +704,8 @@ def hstack(tables, join_type='outer',
         1   3   5   7
         2   4   6   8
     """
+    _check_join_type(join_type, 'hstack')
+
     tables = _get_list_of_tables(tables)  # validates input
     if len(tables) == 1:
```

Comment:
Doesn't really affect the solution, but the reason it failed before is here, that `_get_list_of_tables()` makes a list from a single entry, and then here we just return without checking anything else. The check of `t in ('inner', ...)` does correctly return `False`.

(So, in principle, one could also adjust `_get_list_of_tables`, but I think your solution is better for getting a more helpful error message.)

Reply:
Duh, you're right, brain glitch.